service/arm_templates/main.json

```diff
@@ -50,28 +50,93 @@
     "kubernetesIngressInsecurePort": {
       "type": "int"
     },
+    "masterCount": {
+      "type": "int",
+      "defaultValue": 1
+    },
+    "masterIPAddresses": {
+      "type": "array",
+      "metadata": {
+        "description": "An array containing all the master nodes' ip addresses."
+      }
+    },
+    "masterVMSize": {
```

Comment:
We should provide the VM size via the `azuretpr` custom object. I don't think we need to have the possible values in the template.

For AWS we have configuration in our APIs for the allowed EC2 instance types per installation. We can do the same for Azure.